esp/src/eclwatch/WUQueryWidget.js

```diff
@@ -516,13 +516,15 @@ define([
                         context.downloadToList.set("disabled", true);
                     }
                 });
-                ESPUtil.goToPageUserPreference(this.workunitsGrid, "WUQueryWidget");
-                aspect.after(this.workunitsGrid, "gotoPage", function (deferred, args) {
+                ESPUtil.goToPageUserPreference(this.workunitsGrid, "WUQueryWidget_GridRowsPerPage").then(function(){
+                    context.workunitsGrid.startup();
+                });
+
+                aspect.after(context.workunitsGrid, "gotoPage", function (deferred, args) {
```

Comment:
Move the aspect to before the goToPageUserPreference call.

Reply:
Updated.